src/hooks/useSubStep.js

```diff
@@ -0,0 +1,58 @@
+import {useState, useRef, useCallback} from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = PropTypes.shape({
+    bodyContent: PropTypes.arrayOf(PropTypes.element).isRequired,
+    startFrom: PropTypes.number.isRequired,
+    onFinished: PropTypes.func.isRequired,
+});
```

Comment:
```suggestion
const propTypes = {
    bodyContent: PropTypes.arrayOf(PropTypes.element).isRequired,
    startFrom: PropTypes.number.isRequired,
    onFinished: PropTypes.func.isRequired,
};
```

Shouldn't we do like this?

Reply:
Also, it would be nice to add some description to each prop